SNP/SNPModule.cpp

```diff
@@ -222,70 +173,74 @@ static BOOL __stdcall spi_lock_game_list(int, int, game** out_game_list) {
             joinable = false;
         }
-
         if (ad.game_info.game_state == 12) {
-            // game in progress - these won't be shown anyway but we don't want to initiate p2p
+            // Game in progress - these won't be shown anyway but we don't want to initiate p2p
             joinable = false;
         }
 
-        // agent_state calls ensure_agent so one will be created if needed
+        // agent_state calls ensure_agent so one will be created if needed,
         // this kicks off the p2p connection process
         if (joinable) {
             switch (g_crown_link->juice_manager().lobby_agent_state(ad)) {
                 case JUICE_STATE_CONNECTING: {
-                    prefixes += "...";
-                    prefixes += std::format("{}", char(ColorByte::Gray));
+                    ss << ColorByte::Gray << "...";
                 } break;
                 case JUICE_STATE_FAILED: {
-                    prefixes += " !! ";
-                    prefixes += std::format("{}", char(ColorByte::Gray));
+                    ss << ColorByte::Gray << " !! ";
                     g_crown_link->juice_manager().send_connection_request(ad.game_info.host);
                 } break;
                 case JUICE_STATE_DISCONNECTED: {
-                    prefixes += "-";
-                    prefixes += std::format("{}", char(ColorByte::Gray));
+                    ss << ColorByte::Gray << "-";
                     g_crown_link->juice_manager().send_connection_request(ad.game_info.host);
                 } break;
                 case JUICE_STATE_CONNECTED:
                 case JUICE_STATE_COMPLETED: {
                     switch (g_crown_link->juice_manager().final_connection_type(ad.game_info.host)) {
                         case JuiceConnectionType::Relay: {
-                            prefixes += "[R]";
+                            ss << ColorByte::Gray << "[R]";
                         } break;
                         case JuiceConnectionType::Radmin: {
-                            prefixes += std::format("[Radmin {}]", char(131));
+                            ss << ColorByte::Gray << std::format("[Radmin {}]", static_cast<char>(131));  // char :(
+                        } break;
+                        default: {
+                            ss << ColorByte::Gray << static_cast<char>(187);  // char >>
                         } break;
-                        default:
-                            prefixes += std::format("{}", char(187));
                     }
                 }
             }
         }
 
-        update_lobby_name(ad, prefixes, joinable);
+        if (ad.original_name.empty()) {
+            ad.original_name = ad.game_info.game_name;
+        }
+        if (want_space) {
+            ss << " ";
+        }
+        ss << ColorByte::Gray << ad.original_name;
+
+        if (snp_config.add_map_to_lobby_name) {
+            ss << (joinable ? ColorByte::Blue : ColorByte::Revert) << ad.game_info.game_description;
```

Comment:
The game description is a long string that has much more than just the map name. You removed all the logic that just finds the map name.

Example game description: ` ,43,,6,,1e,,1,cb2edaab,6,,Jesse\rCold Sweat\r\n`

Also this never truncates the player name so if a player name is long then the map name could get completely cut off. I don't necessarily think that's a bad thing but it's a change in behavior we should discuss rather than just arbitrarily change.

Reply:
Extracting the string was not at all documented, no comments, no function name.

Player name should always be fully visible.

Reply:
Ok, I'll yield on that.